src/Microsoft.TemplateEngine.Cli/TemplateInvoker.cs

```diff
@@ -202,50 +196,37 @@ private async Task<NewCommandStatus> CreateTemplateAsync(ITemplateInfo template,
                         }
                     }
 
-                    if (!string.IsNullOrEmpty(template.ThirdPartyNotices))
+                    if (!string.IsNullOrEmpty(templateArgs.Template.ThirdPartyNotices))
                     {
-                        Reporter.Output.WriteLine(string.Format(LocalizableStrings.ThirdPartyNotices, template.ThirdPartyNotices));
+                        Reporter.Output.WriteLine(string.Format(LocalizableStrings.ThirdPartyNotices, templateArgs.Template.ThirdPartyNotices));
                     }
 
-                    return HandlePostActions(instantiateResult, commandInput);
+                    return HandlePostActions(instantiateResult, templateArgs);
                 case CreationResultStatus.CreateFailed:
                     Reporter.Error.WriteLine(string.Format(LocalizableStrings.CreateFailed, resultTemplateName, instantiateResult.ErrorMessage).Bold().Red());
                     return NewCommandStatus.CreateFailed;
-                //TODO: needed?
-                //case CreationResultStatus.MissingMandatoryParam:
-                //    if (!string.IsNullOrWhiteSpace(instantiateResult.ErrorMessage))
-                //    {
-                //        // TODO: rework to avoid having to reparse.
-                //        // The canonical info could be in the ITemplateMatchInfo, but currently isn't.
-                //        TemplateCommandInput reparsedCommand = TemplateCommandInput.ParseForTemplate(template, commandInput, _hostDataLoader.ReadHostSpecificTemplateData(template));
-                //        IReadOnlyList<string> missingParamNamesCanonical = instantiateResult.ErrorMessage.Split(new[] { ',' })
-                //            .Select(x => reparsedCommand.VariantsForCanonical(x.Trim())
-                //                                        .DefaultIfEmpty(x.Trim()).First())
-                //            .ToList();
-                //        string fixedMessage = string.Join(", ", missingParamNamesCanonical);
-                //        Reporter.Error.WriteLine(string.Format(LocalizableStrings.MissingRequiredParameter, fixedMessage, resultTemplateName).Bold().Red());
-                //    }
-                //    return New3CommandStatus.MissingMandatoryParam;
+                //this is unlikely case as these errors are caught on parse level now
```

Comment:
What does "unlikely" means here? Never(should we throw exception)?

Reply:
previously we did validation only for choice parameters on CLI level and didn't check if mandatory parameters are specified. 

Now we do validation for all kinds of parameters on the parser level. So if we do job right, this should not happen. But in case logic of validation diverges in parser and core we might end up here.  

Imo it's best to display core error here as we don't know the reason. Possible improvement is to separate invalid parameters and reason in Edge. Then we can do display reason, and convert canonical (name in json) parameter name to alias not to confuse user with canonical name. Edge doesn't know the alias name (this is CLI specific).